### ISSUE TITLE
RealisticBiomes loses connection to database

### ISSUE THREAD

**Issue by drjawa** [Civcraft/RealisticBiomes#10]
This has been needed for some time. If we are going to do sharding, then we will need to get this out of the way anyway so consider normal priority as part of sharding development.

**Comment by ttk2** [comment]
agreed. Also we might want to consider moving it to Reddis, probably not
worth it currently, but if there are issues with the ram cache, Reddis is a
keystore, so just use the chunk chords as the key and plants as the
content, since reddis ram caches it will handle all of that, with no other
dependencies it would be an easy conversion and might some day be easier
than improving the ram cache if needed.

**Comment by erocs** [comment]
You'd still have the latency of talking to the reddis server vs. in-proc memory access. You would also need to setup reddis to persist this storage. And that's another potential data store you need to backup and maintain.

**Comment by ttk2** [comment]
good points, the additional latency is probably reason enough alone. Reddis
does not persist by default?

**Comment by erocs** [comment]
It does... every 5 minutes by default with RDB. You'd want to enable both persistence layers, AOF and RDB, since AOF gives the ability to save every second. Then you'd backup the RDB unless there was a reddis/VM crash and the AOF needed to be used.

**Comment by ttk2** [comment]
every 5 minutes sounds pretty ok, one second saving might be a bit much.

**Comment by plebes** [comment]
Is this still a bug? If so, is it for certain a problem with the RB plugin and not the database or other part of the system?

**Comment by ttk2** [comment]
We have not gotten it in a while. I think it's fixed. Yes we know it was RB
connections expire and its the plugins job go reopen them if they do.

**Comment by erocs** [comment]
It should be the connection pool's job to re-open the connection. RB
doesn't use one though.

**Comment by ProgrammerDan** [comment]
Yeah I was ranting to Gipsy about the RB database code a few hours back.
It's pretty bad, but if it's working semi-stable then not priority atm.

**Comment by ttk2** [comment]
An open source connection pool setup would probably be nice across all
plugins.

**Comment by plebes** [comment]
Suggestions? BoneCP?

**Comment by erocs** [comment]
I was originally thinking C3P0 but that looks like it would work too.

**Comment by ttk2** [comment]
just pick one that works and run with it.

**Comment by rourke750** [comment]
Just put in in CivModCore since thats what its used for...
